sphinx/builders/__init__.py

```diff
@@ -361,6 +362,9 @@ def build(self, docnames, summary=None, method='update'):
         else:
             if method == 'update' and not docnames:
                 logger.info(bold('no targets are out of date.'))
+                self.interrupt_tasks = SerialTasks()
+                self.interrupt()
+                self.interrupt_tasks.join()
```

Comment:
I think these process is not "interrupted". So I feel the naming is strange.
In addition, `Builder.interrupt()` looks like the builder goes to interrupt the process because of the name. But, in actual, this is a handler on skipping build. So I feel also strange.
(I know another handler on finished is named as `Builder.finish()`.)

So I feel another naming is good to this case. How about adding a parameter to `Builder.finish()` (like `status`)?

Note: in future, IMO, it would be nice if we can rename `Builder.finish()` also.

Reply:
Anyway, we should not use the word "interrupt" for this case. This is not interruption.